### PR TITLE
Improve top-level help for `uv tool` commands

### PR DESCRIPTION
**Pull request by zanieb**
More work needs to be done for all of the options

### CODE REVIEW

File: crates/uv-cli/src/lib.rs [hunk 1]
```diff
@@ -2606,30 +2606,83 @@ pub struct ToolNamespace {
...
+    /// The name of the command can include an exact version in the format
+    /// `<package>@<version>`, e.g., `uv run ruff@0.3.0`. If more complex
+    /// version specification is desired or if the command is provided by a
+    /// different package, use `--from`.
+    ///
+    /// If the tool was previously installed, i.e., via `uv tool install`, the
+    /// installed version will be used unless a version is requested or the
+    /// `--isolated` flag is used.
+    ///
+    /// A `uvx` alias is provided for `uv tool run` for quick invocations.
```
Comment (charliermarsh) [inline crates/uv-cli/src/lib.rs:2622]: I might say... uvx is provided as a convenient alias for uv tool run.

File: crates/uv-cli/src/lib.rs [hunk 2]
```diff
@@ -2606,30 +2606,83 @@ pub struct ToolNamespace {
...
+    ///
+    /// If no command is provided, the installed tools are displayed.
+    ///
+    /// Packages are installed into an ephemeral virtual environment in the uv
+    /// cache directory.
+    ///
+    /// Be careful when invoking tools, package installation can execute
```
Comment (charliermarsh) [inline crates/uv-cli/src/lib.rs:2629]: "when invoking tools, as package installation..."

(Is this piece necessary?)

**Review by charliermarsh — APPROVED** [review]

File: crates/uv-cli/src/lib.rs (hunk 2 above)
Comment (zanieb) [reply crates/uv-cli/src/lib.rs:2629]: I'm a little hesitant on it, but I want to warn about dependency confusion attacks e.g. `uvx foobar` when `foobar` is actually provided by an official `foobar-cli` is a real risk.

File: crates/uv-cli/src/lib.rs (hunk 1 above)
Comment (zanieb) [reply crates/uv-cli/src/lib.rs:2622]:
```suggestion
    /// `uvx` is provided as a convenient alias for `uv tool run`, their behavior is identical.
```

**Pull request merged**